OneSignalSDK/onesignal/src/main/java/com/onesignal/OSSessionManager.java

```diff
@@ -0,0 +1,243 @@
+            return notificationIds != null && notificationIds.length() > 0;
+        }
+
+        public static class Builder {
+
+            private JSONArray notificationIds;
+            private Session session;
+
+            public static SessionResult.Builder newInstance() {
+                return new SessionResult.Builder();
+            }
+
+            private Builder() {
+            }
+
+            public SessionResult.Builder setNotificationIds(JSONArray notificationIds) {
+                this.notificationIds = notificationIds;
+                return this;
+            }
+
+            public SessionResult.Builder setSession(Session session) {
+                this.session = session;
+                return this;
+            }
+
+            public SessionResult build() {
+                return new SessionResult(this);
+            }
+        }
+    }
+
+    interface SessionListener {
+        void onSessionRestarted();
+    }
+
+    public enum Session {
+        DIRECT,
+        INDIRECT,
+        UNATTRIBUTED,
+        DISABLED,
+        ;
+
+        public static @NonNull
+        Session fromString(String value) {
+            if (value == null || value.isEmpty())
+                return UNATTRIBUTED;
+
+            for (Session type : Session.values()) {
+                if (type.name().equalsIgnoreCase(value))
+                    return type;
+            }
+            return UNATTRIBUTED;
+        }
+    }
+
+    private Session session = null;
+    private String notificationId = null;
+    private JSONArray notificationIds = null;
+    private Long sessionTime = null;
+    private SessionListener sessionListener = null;
+
+    public OSSessionManager() {
+    }
+
+    public OSSessionManager(SessionListener sessionListener) {
+        this.sessionListener = sessionListener;
+    }
+
+    void addSessionNotificationsIds(JSONObject jsonObject) {
+        if (session == null || session == Session.UNATTRIBUTED)
+            return;
+        try {
+            if (notificationId != null && session == Session.DIRECT) {
+                jsonObject.put(DIRECT_TAG, true);
+                jsonObject.put(OutcomesUtils.NOTIFICATIONS_IDS, new JSONArray().put(notificationId));
+            } else {
+                if (notificationIds == null)
+                    setLastNotificationsId();
+                if (notificationIds.length() > 0 && session == Session.INDIRECT) {
+                    jsonObject.put(DIRECT_TAG, false);
+                    jsonObject.put(OutcomesUtils.NOTIFICATIONS_IDS, notificationIds);
+                }
+            }
+        } catch (JSONException e) {
+            OneSignal.Log(OneSignal.LOG_LEVEL.ERROR, "Generating addNotificationId:JSON Failed.", e);
+        }
+    }
+
+    void cleanSession() {
+        this.session = null;
+    }
+
+    void restartSessionIfNeeded() {
+        long resetTime = sessionTime != null ? System.currentTimeMillis() - sessionTime : HALF_MIN_IN_MILLIS;
```

Comment:
This is duplicated code, we check this in `OneSignal.doSessionInit` where this is being called from

Reply:
@Jeasmine is there a reason we check session time twice? Can we remove this if statement since we check this in the previous method call?

Reply:
This isn't relevant after the recent changes we have made